src/test/java/org/broadinstitute/hellbender/engine/ReferenceFileSourceUnitTest.java

```diff
@@ -12,7 +16,61 @@
 
     @Test(expectedExceptions = UserException.MissingReference.class)
     public void testMissingReferenceFile() throws IOException {
-        new ReferenceFileSource(GATKBaseTest.getSafeNonExistentFile("NonExistentReference.fasta").getAbsolutePath());
+        new org.broadinstitute.hellbender.engine.datasources.ReferenceFileSource(
+                GATKBaseTest.getSafeNonExistentFile("NonExistentReference.fasta")
+                        .getAbsolutePath());
+    }
+
+    @Test(expectedExceptions = UserException.MissingReferenceFaiFile.class)
+    public void testReferenceInPathWithoutFai() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path refPath = jimfs.getPath("reference.fasta");
+            Files.createFile(refPath);
+            final Path dictPath = jimfs.getPath("reference.dict");
+            Files.createFile(dictPath);
+
+            new org.broadinstitute.hellbender.engine.ReferenceFileSource(refPath);
+        }
+    }
+
+    @Test(expectedExceptions = UserException.MissingReferenceDictFile.class)
+    public void testReferenceInPathWithoutDict() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path refPath = jimfs.getPath("reference.fasta");
+            Files.createFile(refPath);
+            final Path faiFile = jimfs.getPath("reference.fasta.fai");
+            Files.createFile(faiFile);
+
+            new org.broadinstitute.hellbender.engine.ReferenceFileSource(refPath);
+        }
+    }
+
+    @Test
+    public void testReferenceInPath() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path refPath = jimfs.getPath("reference.fasta");
+            Files.createFile(refPath);
+            final Path faiFile = jimfs.getPath("reference.fasta.fai");
+            Files.createFile(faiFile);
+            final Path dictPath = jimfs.getPath("reference.dict");
+            Files.createFile(dictPath);
+
+            new org.broadinstitute.hellbender.engine.ReferenceFileSource(refPath);
```

Comment:
Could you copy a real fasta into jimfs and then assert that we can actually get correct data out of it instead of just that it doesn't crash?

Reply:
Is there a reference test that reads from disk that I can use as a model?